first-party/jib-layer-filter-extension-maven/src/test/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtensionTest.java

```diff
@@ -291,4 +340,150 @@ public void testExtendContainerBuildPlan_complex() throws JibPluginExtensionExce
         Arrays.asList("/alpha/Bob", "/beta/Bob", "/gamma/Bob"), layerToExtractionPaths(newLayer5));
     assertEquals(Arrays.asList("/gamma/Charlie"), layerToExtractionPaths(newLayer6));
   }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noProjectDependenciesResolver() {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+
+    when(config.isCreateParentDependencyLayers()).thenReturn(true);
+    try {
+      new JibLayerFilterExtension()
+          .extendContainerBuildPlan(buildPlan, null, Optional.of(config), mavenData, logger);
+      fail();
+    } catch (JibPluginExtensionException ex) {
+      assertEquals(JibLayerFilterExtension.class, ex.getExtensionClass());
+      assertEquals(
+          "Try to get parent dependencies, but ProjectDependenciesResolver is null. Please use a more recent jib plugin version to fix this.",
+          ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noParent() {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+
+    when(config.isCreateParentDependencyLayers()).thenReturn(true);
+    when(mavenProject.getParent()).thenReturn(null);
+
+    try {
+      JibLayerFilterExtension extension = new JibLayerFilterExtension();
+      extension.dependencyResolver = projectDependenciesResolver;
```

Comment:
I think setting this is unnecessary for the test.

Reply:
ok